xonsh/shell.py

```diff
@@ -115,19 +117,18 @@ def __getattr__(self, attr):
         """Delegates calls to appropriate shell instance."""
         return getattr(self.shell, attr)
 
-    def _init_environ(self, ctx, config, rc):
+    def _init_environ(self, ctx, config, rc, scriptcache, cacheall):
         self.execer = Execer(config=config, login=self.login)
+        self.execer.scriptcache = scriptcache
+        self.execer.cacheall = cacheall
+        env = builtins.__xonsh_env__
         if ctx is None:
-            builtins.__xonsh_ctx__ = self.ctx = context = {}
-            if self.login or self.stype != 'none':
-                # load xontrib files listed in the config file
-                names = builtins.__xonsh_config__.get('xontribs', ())
-                for name in names:
-                    xontribs.update_context(name, ctx=context)
-                # load run contol files
-                env = builtins.__xonsh_env__
+            if self.stype == 'none' and not self.login:
+                self.ctx = {}
+            else:
```

Comment:
It looks like this got rid of all of the `xontrib` loading accidentally on merge